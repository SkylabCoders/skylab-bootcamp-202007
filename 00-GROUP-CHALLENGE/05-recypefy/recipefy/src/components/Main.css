.main__section {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
}

.Main-page {
  width: 100%;
}

main {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  width: 100%;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
}

@media screen and (min-width: 1024px) {
  main {
    -webkit-box-align: end;
        -ms-flex-align: end;
            align-items: flex-end;
  }
  .loading-container {
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    width: 62%;
    -webkit-box-orient: vertical;
    -webkit-box-direction: normal;
        -ms-flex-direction: column;
            flex-direction: column;
    margin: auto;
    -webkit-box-align: center;
        -ms-flex-align: center;
            align-items: center;
    height: 80vw;
  }
}
/*# sourceMappingURL=Main.css.map */