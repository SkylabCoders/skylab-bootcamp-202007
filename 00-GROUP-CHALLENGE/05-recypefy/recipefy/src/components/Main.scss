.main__section {
	display: flex;
}

.Main-page {
	width: 100%;
}

main {
	display: flex;
	flex-direction: column;
	width: 100%;
	align-items: center;
}


@media screen and (min-width: 1024px) {
	main {
		align-items: flex-end;
	}

	.loading-container {
		display: flex;
		width: 62%;
		flex-direction: column;
		margin: auto;
		align-items: center;
		height: 80vw;
	}
}