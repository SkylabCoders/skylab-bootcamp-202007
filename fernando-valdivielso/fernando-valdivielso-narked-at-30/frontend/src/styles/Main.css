/* Desktop */

@media screen and (min-width: 992px) {
	.main {
		display: flex;
		justify-content: row;
		position: relative;
	}

	.main__diver img {
		width: 50vw;
		height: 100vh;
	}

	.main__kittiwake img {
		width: 50vw;
		height: 100vh;
	}
}
