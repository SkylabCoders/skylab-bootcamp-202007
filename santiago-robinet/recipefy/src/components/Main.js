import React from 'react';


function Main(){
    return(
        <>
          <p> I'm da Main!! And I work!!</p>
        </>
    );
}

export default Main;