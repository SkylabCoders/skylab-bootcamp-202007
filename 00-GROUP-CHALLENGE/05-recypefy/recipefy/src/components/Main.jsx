import React, { useState, useEffect } from 'react';
import ProfileComponent from './ProfileComponent';
import Recomendation from './Recomendation';
import TopFiveComponent from './TopFiveComponent';
import PreferencesList from './PreferenceList';
import { loadRecipe } from '../actions/RecipeAction';
import recipeStore from '../stores/RecipeStore';
import './Main.scss';

function Main() {
	const [recipeList, setRecipeList] = useState(recipeStore.getRecipes);
	const [chargeTime, setChargeTime] = useState(false);

	useEffect(() => {
		recipeStore.addChangeListener(onChange);
		if (recipeList.length === 0) {
			loadRecipe();
			chargeList();
		} else {
			resetTime();
		}
		return () => recipeStore.removeChangeListener;
	}, [recipeList.length, chargeTime]);

	function onChange() {
		setRecipeList(recipeStore.getRecipes);
	}
	function chargeList() {
		setTimeout(() => {
			setChargeTime(true);
		}, 3000);
	}
	function resetTime() {
		setChargeTime(false);
	}

	return (
		<div className="body-content">
			{recipeList.length !== 0 && (
				<>
					<ProfileComponent />
					<main className="Main-page">
						<Recomendation />
						<TopFiveComponent />
						<PreferencesList />
					</main>
				</>
			)}
			{recipeList.length === 0 && (
				<div className="result-section body-content">
					{!chargeTime && (
						<div className="loading-container">
							<img
								className="loading"
								alt="loading"
								src="https://media1.tenor.com/images/8a1df0caea44830aa0ecf99ac223c747/tenor.gif?itemid=15085886"
							></img>
							<p>loading...</p>
						</div>
					)}
					{chargeTime && (
						<>
							<h2 className="box__title">Ups! We have and error</h2>
							<h3>Try again later</h3>
							<p>We have some isues wit the data base</p>
						</>
					)}
				</div>
			)}
		</div>
	);
}

export default Main;
