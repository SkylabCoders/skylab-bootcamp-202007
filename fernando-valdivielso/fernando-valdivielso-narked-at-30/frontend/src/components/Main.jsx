import React from 'react';
import diver from '../images/diver.jpg';
import kittiwake from '../images/kittiwake.jpg';
import '../styles/Main.css';

function Main() {
	return (
		<div className='main'>
			<div className='main__diver'>
				<img src={diver} alt='diver' />
			</div>
			<nav className='header__navigation'>
				<a href='/diver' className='header__navigation__button'>
					<span className='button__text'>Divers</span>
					<div className='button__text__wave'></div>
				</a>

				<a href='/dive-site' className='header__navigation__button'>
					<span className='button__text'>Dive Sites</span>
					<div className='button__text__wave'></div>
				</a>
			</nav>
			<div className='main__kittiwake'>
				<img src={kittiwake} alt='wreck' />
			</div>
		</div>
	);
}

export default Main;
