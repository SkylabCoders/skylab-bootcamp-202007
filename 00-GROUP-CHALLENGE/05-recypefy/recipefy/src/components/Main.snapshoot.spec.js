import renderer from 'react-test-renderer';
import Main from './Main';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
describe('Main snapshot', () => {
    const tree = renderer.create(
        <BrowserRouter>
            <Main />
        </BrowserRouter>
    );
    it('should match', () => {
        expect(tree.toJSON()).toMatchSnapshot();
    })
})