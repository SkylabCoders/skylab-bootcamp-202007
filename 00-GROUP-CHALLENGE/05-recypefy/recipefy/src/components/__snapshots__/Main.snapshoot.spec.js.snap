// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main snapshot should match 1`] = `
<div
  className="body-content"
>
  <div
    className="result-section body-content"
  >
    <div
      className="loading-container"
    >
      <img
        alt="loading"
        className="loading"
        src="https://media1.tenor.com/images/8a1df0caea44830aa0ecf99ac223c747/tenor.gif?itemid=15085886"
      />
      <p>
        loading...
      </p>
    </div>
  </div>
</div>
`;
